torch/csrc/api/include/torch/data/datasets/chunk.h

```diff
@@ -377,6 +432,7 @@ class ChunkDataset final
   // provide a references to chunk sampler. Used mainly in distributed data
   // loading to set the epoch number for the sampler.
   ChunkSamplerType& chunk_sampler() {
+    std::lock_guard<std::mutex> lock(chunk_index_guard_);
```

Comment:
This lock is not useful. What is your expectation on this lock.

Reply:
It seems you are looking at the old version. Could you check if you have access to the update?